gossip/minimal/goshawk.go

```diff
@@ -173,30 +156,28 @@ func (hawk *Goshawk) Fly(ctx context.Context) {
 			glog.Infof("finished Checker(%s)", origin.Name)
 		}(origin)
 	}
+
 	wg.Wait()
 }
 
 // Scanner runs a continuous scan of the destination log.
 func (hawk *Goshawk) Scanner(ctx context.Context) {
-	ticker := time.NewTicker(hawk.dest.MinInterval)
-	for {
-		glog.V(1).Infof("Scanner(%s): run scan from %d to current size", hawk.dest.Name, hawk.scanOpts.StartIndex)
-		s := scanner.NewScanner(hawk.dest.Log, hawk.scanOpts)
-		size, err := s.ScanLog(ctx, hawk.foundCert, foundPrecert)
-		if err != nil {
-			glog.Errorf("Scanner(%s) terminated: %v", hawk.dest.Name, err)
-			return
-		}
-		hawk.scanOpts.StartIndex = size
-
-		glog.V(2).Infof("Scanner(%s): wait for a %s tick", hawk.dest.Name, hawk.dest.MinInterval)
-		select {
-		case <-ctx.Done():
-			glog.Infof("Scanner(%s): termination requested", hawk.dest.Name)
-			return
-		case <-ticker.C:
+	fetcher := scanner.NewFetcher(hawk.dest.Log, &hawk.fetchOpts)
+	fetcher.Run(ctx, func(batch scanner.EntryBatch) {
+		glog.V(2).Infof("Scanner(%s): examine batch [%d, %d)", hawk.dest.Name, batch.Start, int(batch.Start)+len(batch.Entries))
+		for i, entry := range batch.Entries {
+			index := batch.Start + int64(i)
+			rawLogEntry, err := ct.RawLogEntryFromLeaf(index, &entry)
+			if rawLogEntry == nil {
```

Comment:
Are you sure you don't want to check `err != nil` here? Maybe check both?

Reply:
I don't see the point -- `RawLogEntryFromLeaf` complies with normal Go conventions and always returns a `nil` entry if `err` is non-nil, and the key check for the code below is whether there's a `RawLogEntry` to play with.

Reply:
Again, feel free. I just got used to see `if err != nil` first thing after any call (precisely for the same "conventions" reason), so seeing something different triggers "why?"s in my head automatically. Also, some linters complain when `err` is ignored.

Reply:
Added `err != nil` arm too.